models/generators.py

```diff
@@ -150,24 +150,21 @@ def call(self, embedding, noise):
         x = self.deconv2d_4(x)
         x = self.conv2d_4(x)
 
-        x = tanh(x)
+        x = self.tanh(x)
 
         return x, mean, log_sigma
 
     def loss(self, predictions_on_fake, mean, log_sigma):
         """ Only calculate the loss based on the discriminator
             predictions for the images generated by this model.
         """
-        kl_coeff = 2
-        loss = tf.nn.sigmoid_cross_entropy_with_logits(
-            labels=tf.ones_like(predictions_on_fake),
-            logits=predictions_on_fake
-        )
-        loss = tf.reduce_mean(loss) + kl_coeff * self.kl_loss(mean, log_sigma)
+        kl_coeff = 1
+        loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(labels=tf.ones_like(predictions_on_fake), logits=predictions_on_fake))
+        loss = loss + kl_coeff * self.kl_loss(mean, log_sigma)
         return loss
 
     def kl_loss(self, mean, log_sigma):
-        loss = -log_sigma + .5 * (-1 + tf.exp(2. * log_sigma) + tf.square(mean))
+        loss = -.5 * log_sigma + .5 * (-1 + tf.exp(2. * log_sigma) + tf.math.square(mean))
```

Comment:
You could have just moved this into the bracket but it doesn't matter :stuck_out_tongue_closed_eyes: